src/service.py

```diff
@@ -49,15 +51,21 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         self.snap_client.set(snap_config, typed=True)
 
 
-def snap_install(resource: str) -> SnapService:
+def snap_install(resource: Optional[str]) -> SnapService:
     """Install or refresh the snap, and return the snap service.
 
+    If resource is passed it will have preference over the one in the snapcraft store.
+
     Raises an exception on error.
     """
     try:
         logger.debug("installing snap.")
-        logger.debug("fetching from resource.")
-        snap_client = snap.install_local(resource, dangerous=True)
+        if resource:
+            logger.debug("fetching %s from resource.", SNAP_NAME)
```

Comment:
```suggestion
            logger.debug("installing %s snap from resource.", SNAP_NAME)
```

"fetching" doesn't really match what is happening.